trafilatura/deduplication.py

```diff
@@ -4,25 +4,28 @@
 # 3.11+: from typing import Self
 
 import re
+import sys
 import string
-
 from difflib import SequenceMatcher
 from functools import lru_cache
 from hashlib import blake2b
 from operator import add
 from threading import RLock
 from typing import Any, Dict, List, Optional, Union
 
+import unicodedata
 from lxml.etree import _Element
 
 from .settings import LRU_SIZE
 from .utils import trim
 
-
 STRIP_EXTENSION = re.compile(r"\.[^/?#]{2,63}$")
 
 BIN_COUNT_FUNC = getattr(int, "bit_count", lambda x: bin(x).count("1"))
 
+PUNCT_TBL = dict.fromkeys((i for i in range(sys.maxunicode)
```

Comment:
`PUNCT_TBL = {i: ' ' for i in range(0x10FFFF) if unicodedata.category(chr(i)).startswith('P')}`

This is simpler and entering the maximum Unicode codepoint manually removes the need for sys.